src/unsatcores/UnsatCore.h

```diff
@@ -0,0 +1,46 @@
+#ifndef OPENSMT_UNSATCORE_H
+#define OPENSMT_UNSATCORE_H
+
+#include <minisat/core/SolverTypes.h>
+#include <minisat/mtl/Vec.h>
+#include <pterms/PTRef.h>
+
+#include <optional>
+
+namespace opensmt {
+
+class PartitionManager;
+
+class UnsatCore {
+public:
+    inline UnsatCore(vec<CRef> && clauses_, PartitionManager const *);
+
+    inline vec<PTRef> const & getTerms() const;
+
+protected:
+    void computeTerms() const;
+
+    vec<CRef> clauses;
+
+    mutable std::optional<vec<PTRef>> termsCache{};
+
+    PartitionManager const * partitionManagerPtr;
```

Comment:
This belongs to the builder in my opinion.

Otherwise, we are still leaking internal details to the outside world.